### PR TITLE
[NFC] Split the new wat parser into multiple files

### PR DESCRIPTION
**Pull request by tlively**
And put the new files in a new source directory, "parser". This is a rough split
and is not yet expected to dramatically improve compile times. The exact
organization of the new files is subject to change, but this splitting should be
enough to make further parser development more pleasant.

### CODE REVIEW

**Comment by tlively** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5960** <a href="https://app.graphite.dev/github/pr/WebAssembly/binaryen/5960" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/WebAssembly/binaryen/5960?utm_source=stack-comment).

File: src/parser/common.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2022 WebAssembly Community Group participants
```
Comment (kripken) [inline src/parser/common.h:2]:
```suggestion
 * Copyright 2023 WebAssembly Community Group participants
```

Comment (kripken) [reply src/parser/common.h:2]: ditto below

**Review by kripken — APPROVED** [review]

File: src/parser/common.h (hunk 1 above)
Comment (kripken) [reply src/parser/common.h:2]: It probably doesn't matter much, but I think for new files 2023 makes more sense. Though if it's just a rename, maybe either way.

**Pull request merged**